File: Editorial/2001-2100/2094_Finding_3_Digit_Even_Numbers/Go/2_Traverse_All_Possible_3_Digit_Even_Numbers.go
func findEvenNumbers(digits []int) []int {
	res := []int{}
	freq := make(map[int]int)
	// Count the number of occurrences of each number in the integer array
	for _, d := range digits {
		freq[d]++
	}
	// Enumerate all three-digit even numbers
	for i := 100; i < 1000; i += 2 {
		freq1 := make(map[int]int)
		num := i
		// Count the frequency of each digit of the current even number
		for num > 0 {
			d := num % 10
			freq1[d]++
			num /= 10
		}
		// Check if the conditions are met
		isValid := true
		for d, count := range freq1 {
			if freq[d] < count {
				isValid = false
				break
			}
		}
		if isValid {
			res = append(res, i)
		}
	}
	return res
}

File: Editorial/2001-2100/2094_Finding_3_Digit_Even_Numbers/Java/2_Traverse_All_Possible_3_Digit_Even_Numbers.java
class Solution {

    public int[] findEvenNumbers(int[] digits) {
        List<Integer> res = new ArrayList<>();
        Map<Integer, Integer> freq = new HashMap<>();
        // Count the number of occurrences of each number in the integer array
        for (int d : digits) {
            freq.put(d, freq.getOrDefault(d, 0) + 1);
        }
        // Enumerate all three-digit even numbers
        for (int i = 100; i < 1000; i += 2) {
            Map<Integer, Integer> freq1 = new HashMap<>();
            int num = i;
            // Count the frequency of each digit of the current even number
            while (num > 0) {
                int d = num % 10;
                freq1.put(d, freq1.getOrDefault(d, 0) + 1);
                num /= 10;
            }
            // Check if the conditions are met
            boolean isValid = true;
            for (Map.Entry<Integer, Integer> entry : freq1.entrySet()) {
                if (freq.getOrDefault(entry.getKey(), 0) < entry.getValue()) {
                    isValid = false;
                    break;
                }
            }
            if (isValid) {
                res.add(i);
            }
        }
        // Convert to array
        int[] result = new int[res.size()];
        for (int j = 0; j < res.size(); j++) {
            result[j] = res.get(j);
        }
        return result;
    }
}
